### PR TITLE
fix(action): use octokit.rest.repos instead of octokit.repos

### PR DESCRIPTION
**Pull request by lucasfernog**
## Motivation

It is wrong on @actions/core v5. We already fixed it on `octokit.issues` usages, but this was missing.

## Approach

Typed the octokit argument and fixed the usage.

### CODE REVIEW

File: packages/action/src/utils.ts [hunk 1]
```diff
@@ -1,5 +1,10 @@
 import fs from "fs";
 import type { ConfigFile, FunctionPipe } from "../../types/src";
+import { Octokit } from '@octokit/core';
+
+declare const GitHub: typeof Octokit & import("@octokit/plugin-rest-endpoint-methods/dist-types/types").Api & {
```
Comment (jbolda) [inline packages/action/src/utils.ts:5]: I think we can improve this, but not 100% so looking into it.

**Review by jbolda — CHANGES_REQUESTED** [review]

**Review by jbolda — APPROVED** [review]
Better typing ended up exposing the fact that we missed updating the mocks in the tests. So I think we have it all squared away now.

File: packages/action/src/utils.ts [hunk 2]
```diff
@@ -208,15 +193,15 @@ export const createReleases = curry(
...
           );
-          const uploadedAsset = await octokit.repos
+          const uploadedAsset = await octokit.rest.repos
             .uploadReleaseAsset({
               owner,
               repo,
               release_id: releaseResponse.id,
               name: asset.name,
-              data: fs.readFileSync(asset.path),
+              data: fs.readFileSync(asset.path, { encoding: "utf-8" }),
```
Comment (lucasfernog) [inline packages/action/src/utils.ts:202]: We can't use a string here, we need to keep the old code. See https://github.com/tauri-apps/tauri-action/pull/45

Comment (jbolda) [reply packages/action/src/utils.ts:202]: Oooh, good catch. I bet that was the poor types pushing it towards that.

Comment (jbolda) [reply packages/action/src/utils.ts:202]: Hmm, even with their types directly it asks for a string.

**Pull request merged**